core/exchange.go

```diff
@@ -177,7 +178,11 @@ func (ce *Exchange) getExtendedHeaderByHeight(ctx context.Context, height *int64
 	if err != nil {
 		return nil, fmt.Errorf("extending block data for height %d: %w", b.Header.Height, err)
 	}
-	// create extended header
+
+	// TODO(@Wondertan): This is a hack to deref Data, allowing GC to pick it up.
+	//  The better footgun-less solution is to change core.ResultSignedBlock fields to be pointers instead of values.
+	b.Data = types.Data{}
+
 	eh, err := ce.construct(&b.Header, &b.Commit, &b.ValidatorSet, eds)
```

Comment:
```suggestion

	// copying the commit and validator set so we break the link with
	// the signed header and the signed block is garbage collected
	commit := &types.Commit{
		Height:     b.Commit.Height,
		Round:      b.Commit.Round,
		BlockID:    b.Commit.BlockID,
		Signatures: b.Commit.Signatures,
	}
	vs := &types.ValidatorSet{
		Validators: b.ValidatorSet.Validators,
		Proposer:   b.ValidatorSet.Proposer,
	}

	// create extended header
	eh, err := ce.construct(&b.Header, commit, vs, eds)
```

This fixes the bug because the commit and validator set are used as they are in:
https://github.com/celestiaorg/celestia-node/blob/353141fb5a0a33c10a8678ac6e92a90b02f74dbd/header/header.go#L64-L70

Which results in GC not liberating that memory. 

After copying their values as per the suggestion, the signed block is garbage collected, memory freed, bridge node syncing in under 5GB RAM compared to 50GB RAM.

An alternative solution would be to change: 
https://github.com/celestiaorg/celestia-node/blob/353141fb5a0a33c10a8678ac6e92a90b02f74dbd/header/header.go#L44-L49

To:

```
func MakeExtendedHeader(
	h *core.Header,
	comm core.Commit,
	vals core.ValidatorSet,
	eds *rsmt2d.ExtendedDataSquare,
) (*ExtendedHeader, error) {
```

So the copy is made automatically but I didn't try it as I prefer the clear first approach.

Attached image showing a bridge node syncing a 6s full 8mb blocks network :D 

<img width="1897" alt="image" src="https://github.com/user-attachments/assets/81f9bdfc-68bd-4e4e-9350-7bbd9931bb75">

Reply:
In Slack we agreed to land the proper fix in #3916, while setting Data to be empty is sufficient as a hot fix.